app/src/main/java/com/example/notes/mainactivity/MainActivity.kt

```diff
@@ -38,6 +36,8 @@ class MainActivity : AppCompatActivity() {
         )
 
         setClickOpenItemDetails()
+
+        binding.searchEditText.doAfterTextChanged { activityViewModel.searchNotes(binding.searchEditText.text.toString()) }
```

Comment:
Code looks good! You could extract it in a separate function with a name which describes it's purpose for better code readability, such as setUpSearch()